test/spark/run-gc-test.sh

```diff
@@ -138,11 +138,16 @@ prepare_for_gc() {
   run_lakectl repo create "lakefs://${repo}" "${STORAGE_NAMESPACE}/${repo}/"
   run_lakectl fs upload "lakefs://${repo}/main/not_deleted_file1" -s /local/gc-tests/sample_file
   run_lakectl fs upload "lakefs://${repo}/main/not_deleted_file2" -s /local/gc-tests/sample_file
-  run_lakectl fs upload "lakefs://${repo}/main/not_deleted_file3" -s /local/gc-tests/sample_file
+  local direct_flag_for_non_deleted=''
+  if [[ $LAKEFS_BLOCKSTORE_TYPE == "s3" ]]; then
+    direct_flag_for_non_deleted='--direct'
+  fi
+  run_lakectl fs upload ${direct_flag_for_non_deleted} "lakefs://${repo}/main/not_deleted_file3" -s /local/gc-tests/sample_file
```

Comment:
Shouldn't "lakefs://${repo}/main/not_deleted_file3" be a physical address on s3 outside the storage namespace?

Reply:
The "--direct" flag handles everything under the hood - it does a get+link physical address from the client side.